### PR TITLE
Expose hls manifest walker

### PR DESCRIPTION
**Pull request by grsin**
We wanted to use the manifest walker for another project. We should be able to just expose it in the index.js.

### CODE REVIEW

File: src/walk-manifest.js [hunk 1]
```diff
@@ -114,7 +114,15 @@ var walkPlaylist = function(decrypt, basedir, uri, parent, manifestIndex) {
 		parent.content = new Buffer(parent.content.toString().replace(uri, path.relative(path.dirname(parent.file), manifest.file)));
 	}
 
-  manifest.content = syncRequest('GET', manifest.uri).getBody();
+  res = syncRequest('GET', manifest.uri);
+
+  // Don't throw an error if we get a non 200 just return an emtpy array.
```
Comment (mwhipple) [inline src/walk-manifest.js]: why is this needed? If an error is being returned couldn't this just be handled by a catch block in the calling code?

Comment (mwhipple) [inline src/walk-manifest.js]: why is this needed? If an error is being returned couldn't this just be handled by a catch block in the calling code?

Comment (grsin) [reply src/walk-manifest.js]: That's definitely I guess that's why they handle it that way in `cli.js`. I'll revert it.

**Comment by gkatsev** [comment]
So, looking at the index.js for this, it seems like it basically just calls WalkManifest and then WriteData. The package.json [currently points](https://github.com/videojs/hls-fetcher/blob/master/package.json#L5) at `index.js` rather than `src/index.js`. I think we should make this change regardless of what else we do.

The options to WalkManifest are passed into the `main` method in the file, so, I think we just need an option like `writeData: false` and if it's false it'll return the result of WalkManifest?
Am I overcomplicating things there?

File: src/index.js [hunk 2]
```diff
@@ -8,4 +8,4 @@ var main = function(options) {
   return WriteData(options.decrypt, options.concurrency, resources);
 };
 
-module.exports = main;
+module.exports = {main, WalkManifest};
```
Comment (gkatsev) [inline src/index.js]: actually, this changes the main export, we'd want to do something like
```js
module.exports = main;
module.exports.WalkManifest = WalkManifest;
```

**Comment by gkatsev** [comment]
yeah, probably too complicated so ignore the writeData option :)

**Comment by grsin** [comment]
I'm not exactly sure who the users for the hls-featcher are; my current take is the `writeData:false` option wouldn't really but used by anyone who uses this but I could be wrong.

**Comment by gkatsev** [comment]
the idea was that if you want to only get the walkmanifest resources you set `writeData: false` when you call the method but I think it's not worth it since it means that the method returns different things depending on what you pass in.

File: src/walk-manifest.js (hunk 1 above)
Comment (mwhipple) [reply src/walk-manifest.js]: so I think you were on to something here after all. The current throwing behavior will likely cause an all-or-nothing parse, which means that any bad URLs will lead to an exception which renders the rest of the results unusable (i.e. a bad URL in a rendition playlist will bubble up past the master playlist and any other renditions which may have had all valid URLs). This may or may not be relevant for calling code but I think in our case it is.

I'd advocate the ability to pass an error handler (which by default throws an error) which can be recursively passed through and therefore allow for better collection of errors throughout the tree as desired.

Comment (mwhipple) [reply src/walk-manifest.js]: @gkatsev @BrandonOCasey ^^ thoughts?

Comment (gkatsev) [reply src/walk-manifest.js]: Maybe the error handling can be done as part of the other refactoring PR?

Comment (mwhipple) [reply src/walk-manifest.js]: I think that would be the plan but just wanted to make sure that wasn't any immediate cause for alarm in the approach, thanks!

Comment (gkatsev) [reply src/walk-manifest.js]: yeah, from what I saw in that PR, if we are going with callbacks, I'd expected a node-style callback where errors are returned rather than thrown.
